addons/hr_holidays/models/hr_leave.py

```diff
@@ -427,9 +427,10 @@ def _get_durations(self, check_leave_type=True, resource_calendar=None):
                 days = (leave.date_to - leave.date_from).days + (1 if not leave.request_unit_half else 0.5)
                 public_holidays = self.env['resource.calendar.leaves'].search([
                     ('resource_id', '=', False),
-                    ('calendar_id', '=', calendar.id),
+                    ('calendar_id', 'in', [False, calendar.id]),
                     ('date_from', '<=', leave.date_to),
-                    ('date_to', '>=', leave.date_from)
+                    ('date_to', '>=', leave.date_from),
+                    ('company_id', '=', leave.company_id.id)
```

Comment:
same that calendar, the field isn't required
```suggestion
                    ('company_id', 'in', [False, leave.company_id.id])
```

Reply:
The company field is a readonly field and the default value is the environmental company, so practically its not possible to make the company value as False, unless the user did something in the view or through studio :thinking: 
https://github.com/odoo/odoo/blob/f77b40bba0c92a9a225eb7ade694e79cb804f7bf/addons/resource/models/resource_calendar_leaves.py#L36-L38